src/main.ts

```diff
@@ -14,8 +14,52 @@ import http from 'http'
 import serveStatic from 'serve-static'
 import sourceMapSupport from 'source-map-support'
 
+import pkg from '../package.json'
 import MenuBuilder from './menu'
 
+// Deep linked url
+let deeplinkingUrl: string[] | string
+
+const redirectDeepLinkingUrl = (deepLink: string, m: BrowserWindow) => {
+  if (deepLink.includes(`${pkg.protocolSchemes.native}://`)) {
+    const url = deepLink.split(`${pkg.protocolSchemes.native}://`)[1].split('?')
+
+    if (process.platform == 'win32') {
+      m.webContents.send('deepLink', {
+        view: url[0].slice(0, -1),
+        param: url[1],
+      })
+    } else {
+      m.webContents.send('deepLink', { view: url[0], param: url[1] })
+    }
+  }
+}
+
+// Force Single Instance Application
+const gotTheLock = app.requestSingleInstanceLock()
+if (gotTheLock) {
+  app.on('second-instance', (e, argv) => {
+    // Someone tried to run a second instance, we should focus our window.
+
+    // Protocol handler for win32
+    // argv: An array of the second instance’s (command line / deep linked) arguments
+    if (process.platform == 'win32') {
+      // Keep only command line / deep linked arguments
+      deeplinkingUrl = argv.slice(1)
+    }
+
+    if (mainWindow) {
+      if (mainWindow.isMinimized()) {
+        mainWindow.restore()
+      }
+      mainWindow.focus()
+      redirectDeepLinkingUrl(deeplinkingUrl.toString(), mainWindow)
+    }
+  })
+} else {
+  app.quit()
+}
+
```

Comment:
I think we need to start putting stuff from the main file to a different folder, because `main` becomes a very large and unstable blob of code. Can we move this, let's say, to the `features/deepLinking` ? I'm not sure how right is that, I'll let you decide where it should go, but IMO having it in features is okay.

Also, having these function separately will allow to test them as well

Reply:
I have updated the code structure and move to the deepLinking folder